internal/nodes/ingress_model_rel.go

```diff
@@ -578,11 +583,45 @@ func HTTPRuleToIng(rrname string, namespace string, key string) ([]string, bool)
 		}
 	}
 
-	utils.AviLog.Infof("key: %s, msg: ingresses to compute: %v via httprule %s",
-		key, allIngresses, namespace+"/"+rrname)
+	utils.AviLog.Debugf("key: %s, msg: Ingresses retrieved %s", key, allIngresses)
 	return allIngresses, true
 }
 
+func NsxAlbSettingToGateway(albSettingName string, namespace string, key string) ([]string, bool) {
+	allGateways := make([]string, 0)
+
+	_, err := lib.GetCRDInformers().NsxAlbInfraSettingInformer.Lister().Get(albSettingName)
+	if err != nil {
+		utils.AviLog.Warnf("key: %s, msg: Error getting NsxALbInfraSetting: %v\n", key, err)
+		return nil, false
+	}
+
+	// Get all GatewayClasses from NsxAlbInfraSetting.
+	gwClasses, err := lib.GetSvcAPIInformers().GatewayClassInformer.Informer().GetIndexer().ByIndex(lib.NsxAlbSettingGWClassIndex, "ako.vmware.com/NsxAlbInfraSetting/"+albSettingName)
+	if err != nil {
+		return allGateways, false
+	}
+
+	for _, gwClass := range gwClasses {
+		// Get all Gateways from GatewayClass.
+		gwClassObj, isGwClass := gwClass.(*servicesapi.GatewayClass)
+		if isGwClass {
+			gateways, err := lib.GetSvcAPIInformers().GatewayInformer.Informer().GetIndexer().ByIndex(lib.GatewayClassGatewayIndex, gwClassObj.Name)
+			if err != nil {
+				continue
```

Comment:
should we log the error here? Could be from the indexer or the informer.

Reply:
done